org-roam-db.el

```diff
@@ -492,16 +492,21 @@ If FORCE, force a rebuild of the cache from scratch."
           (let ((contents-hash (org-roam-db--file-hash file)))
             (unless (string= (gethash file current-files)
                              contents-hash)
-              (org-roam--with-temp-buffer file
-               (org-roam-db--clear-file file)
-               (org-roam-db-query
-                [:insert :into files
-                 :values $v1]
-                (vector file contents-hash (list :atime atime :mtime mtime)))
-               (setq file-count (1+ file-count))
-               (when-let ((headlines (org-roam--extract-headlines file)))
-                 (when (org-roam-db--insert-headlines headlines)
-                   (setq headline-count (1+ headline-count)))))))))
+              (condition-case nil
+                  (org-roam--with-temp-buffer file
+                    (org-roam-db--clear-file file)
+                    (org-roam-db-query
+                     [:insert :into files
+                      :values $v1]
+                     (vector file contents-hash (list :atime atime :mtime mtime)))
+                    (setq file-count (1+ file-count))
+                    (when-let ((headlines (org-roam--extract-headlines file)))
+                      (when (org-roam-db--insert-headlines headlines)
+                        (setq headline-count (1+ headline-count)))))
+                (file-error
+                 (setq org-roam-files (remove file org-roam-files))
+                 (org-roam-db--clear-file file)
+                 (message "org-roam-db-build-cache: Skipping unreadable file: %s" file)))))))
```

Comment:
LGTM, but would prefer to warn instead of printing a message:

https://github.com/org-roam/org-roam/blob/2e41c9ef2a197c80e393b4f8b6991cf6a0a91a9b/org-roam-db.el#L264

Reply:
OK, changed the code accordingly. Now epg's error-buffer will be "buried" by the warning buffer, which i consider appropriate in this case.
BTW: The easiest way to see this in action is via `echo abc > org-roam-dir/x.org.gpg` (maybe for multiple files...).